treeherder/services/intermittents_commenter/commenter.py

```diff
@@ -0,0 +1,284 @@
+                                   test_run_count=test_run_count,
+                                   rank=rank,
+                                   priority=priority,
+                                   failure_rate=round(counts['total']/float(test_run_count), 3),
+                                   repositories=counts['per_repository'],
+                                   platforms=counts['per_platform'],
+                                   startday=startday,
+                                   endday=endday.split()[0],
+                                   weekly_mode=self.weekly_mode)
+
+            params = {'comment': {'body': text}}
+            if whiteboard:
+                params['whiteboard'] = whiteboard
+            if change_priority:
+                params['priority'] = change_priority
+
+            if self.test_mode:
+                print(text + '\n')
+            else:
+                self.submit_bug_comment(session, params, bug_id)
+                # sleep between comment submissions to avoid overwhelming servers
+                time.sleep(1)
+
+    def open_file(self, filename, load):
+        with open('treeherder/services/intermittents_commenter/{}'.format(filename), 'r') as myfile:
+            if load:
+                return json.load(myfile)
+            else:
+                return myfile.read()
+
+    def calculate_date_strings(self, mode, numDays):
+        """Returns a tuple of start (in YYYY-MM-DD format) and end date
+           strings (in YYYY-MM-DD HH:MM:SS format for an inclusive day)."""
+
+        yesterday = date.today() - timedelta(days=1)
+        endday = datetime(yesterday.year, yesterday.month, yesterday.day, 23, 59, 59, 999)
+        if mode:
+            startday = yesterday - timedelta(days=numDays)
+        else:
+            # daily mode
+            startday = yesterday
+
+        return startday.isoformat(), endday.strftime('%Y-%m-%d %H:%M:%S.%f')
+
+    def check_whiteboard_status(self, whiteboard):
+        """Extracts stockwell text from a bug's whiteboard status to
+           determine whether it matches specified stockwell text;
+           returns a boolean"""
+
+        stockwell_text = re.search(r'\[stockwell (.+?)\]', whiteboard)
+        if stockwell_text is not None:
+            text = stockwell_text.group(1).split(':')[0]
+            if text == 'fixed' or text == 'disable-recommended' or text == 'infra' or text == 'disabled':
+                return True
+        return False
+
+    def update_whiteboard(self, existing, new):
+        return re.sub('\[stockwell.*?\]', new, existing)
+
+    def new_request(self):
+        session = requests.Session()
+        # Use a custom HTTP adapter, so we can set a non-zero max_retries value.
+        session.mount("https://", requests.adapters.HTTPAdapter(max_retries=3))
+        session.headers = {
+            'User-Agent': settings.TREEHERDER_USER_AGENT,
+            'x-bugzilla-api-key': settings.BUGFILER_API_KEY,
```

Comment:
Could you add `BUGFILER_API_KEY` to settings.py

Reply:
It's already there (that's why it didn't show up in my commit) :)

Reply:
@jmaher and @gbrownmozilla what do you think?

Reply:
Per a convo with gbrown in the IRC channel, we'll stick with the OF commenter account but I'll have the display name updated.